chain/network-primitives/src/metrics.rs

```diff
@@ -6,6 +6,8 @@ use near_metrics::{
 use std::collections::HashMap;
 use strum::VariantNames;
 
+use lazy_static::lazy_static;
```

Comment:
What's the reason for moving this file to network-primitives? It feels to me that those counters should be defined in the network crate itself

Reply:
You're right. Nice catch!

Reply:
Where does the dependency on `Edge` come from? I don't think it is used outside of the network crate

Reply:
So, while `near-client-primitives` only required;

```rust
near_network::{recorder::MetricRecorder, types::{AccountOrPeerIdOrHash, KnownProducer}, PeerInfo}
```

In the case of `near_network::types`, I believed it was safe to extract as much as was possible since that module name typically encapsulates reusable primitives. And that worked fine with the only exception being `Consolidate` which is retained in `near_network::types`.

Following that logic, `routing::Edge` has a number of dependents in `near_network_primitives::types`. So, naturally, it's extracted alongside it.

Reply:
> I believed it was safe to extract as much as was possible since that module name typically encapsulates reusable primitives

Well that is exactly my concern. `Edge` should not be used outside of the network crate. Why do we need `MetricRecorder` in `near-client-primitives`?

Reply:
@bowenwang1996 What would be your suggestion here? I would love to merge this PR first since it already gives net positive result for us. The public interfaces needs more careful thought anyway, it is not like we promoted it to be public here

Reply:
@mfornet is my understanding correct that we don't actually use metric recorder? Can we just get rid of it?

Reply:
> Can we just get rid of it?
Yes 

I was using metric recorder while optimising the network more than a year ago, but it is not used anymore.

Reply:
I see. @miraclx please remove metric_recorder from `NetworkInfoResponse`

Reply:
@mfornet @bowenwang1996 should the whole feature be removed altogether or just drop the field from the `NetworkInfoResponse`? Is the feature still useful without the exposed data?

Reply:
> @mfornet @bowenwang1996 should the whole feature be removed altogether or just drop the field from the NetworkInfoResponse? Is the feature still useful without the exposed data?

I think the feature should be removed altogether, but my point is that we can now revert the change that extracts network primitives since I don't think they should be in a separate crate to begin with @miraclx.

Reply:
`MetricRecorder` was removed, which freed up the dependency on `PeerMessage` which, in-turn, freed up the dependency on `Edge`. This means `::routing` and a number of structures in `::types` we're returned to `near-network`.
But we (`near_client_primitives`) still need `near_network_primitives::types::{AccountOrPeerIdOrHash,  KnownProducer, PeerInfo}`

Reply:
@bowenwang1996 @mfornet Could you review the PR after the update?